vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -808,4 +813,59 @@ ReasonerState getReasonerState() {
 	void setReasonerState(ReasonerState reasonerState) {
 		this.reasonerState = reasonerState;
 	}
+
+	@Override
+	public Correctness writeInferences(OutputStream stream) throws IOException {
+		QueryResult queryAnswer;

```

Comment:
This method should perform the same validations on the reasoner state as `answerQuery `and `exportQueryAnswersToCsv`